libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -299,9 +323,11 @@ void AC_PosControl::get_stopping_point_z(Vector3f& stopping_point) const
 /// init_takeoff - initialises target altitude if we are taking off
 void AC_PosControl::init_takeoff()
 {
-    const Vector3f& curr_pos = _inav.get_position();
+    Vector3f curr_pos;
+    _ahrs.get_relative_position_NED_origin(curr_pos);
```

Comment:
agee, but should get_relative_position_D_origin use baro alt is case of problem like get_relative_position_D_home ?